lock/src/test/java/org/infinispan/lock/impl/lock/RequestExpirationSchedulerTest.java

```diff
@@ -0,0 +1,121 @@
+
+      // We capture the argument so we can run as if the scheduler worked and assert
+      Runnable runnable = argument.getValue();
+      runnable.run();
+
+      assertTrue(booleanCompletableFuture.isDone());
+      assertFalse(booleanCompletableFuture.get());
+      assertEquals(0, expirationScheduler.countScheduledRequests());
+   }
+
+   @Test
+   public void testScheduleForCompletionAddsASingleRequestById() throws Exception {
+
+      expirationScheduler.scheduleForCompletion("123", booleanCompletableFuture, 100, TimeUnit.SECONDS);
+      expirationScheduler.scheduleForCompletion("123", booleanCompletableFuture, 0, TimeUnit.SECONDS);
+      expirationScheduler.scheduleForCompletion("123", booleanCompletableFuture, 50, TimeUnit.SECONDS);
+
+      RequestExpirationScheduler.ScheduledRequest scheduledRequest = expirationScheduler.get("123");
+
+      assertEquals(scheduledFutureMock, scheduledRequest.getScheduledFuture());
+      assertEquals(booleanCompletableFuture, scheduledRequest.getRequest());
+      assertEquals(1, expirationScheduler.countScheduledRequests());
+      verify(scheduledExecutorService).schedule(any(Runnable.class), eq(100L), eq(TimeUnit.SECONDS));
+      verify(scheduledExecutorService, never()).schedule(any(Runnable.class), eq(0L), eq(TimeUnit.SECONDS));
+      verify(scheduledExecutorService, never()).schedule(any(Runnable.class), eq(50L), eq(TimeUnit.SECONDS));
+   }
+
+   @Test
+   public void testCompletedRequestsShouldNotBeScheduled() throws Exception {
+      CompletableFuture<Boolean> request = new CompletableFuture();
+      request.complete(true);
+
+      expirationScheduler.scheduleForCompletion("123", request, 10, TimeUnit.MILLISECONDS);
+      assertEquals(0, expirationScheduler.countScheduledRequests());
+   }
+
+   @Test
+   public void testAbortSchedulingWithCompletedRequest() throws Exception {
+
+      expirationScheduler.scheduleForCompletion("123", booleanCompletableFuture, 42, TimeUnit.SECONDS);
+      RequestExpirationScheduler.ScheduledRequest scheduledRequest = expirationScheduler.get("123");
+      assertEquals(1, expirationScheduler.countScheduledRequests());
+
+      booleanCompletableFuture.complete(true);
+
+      expirationScheduler.abortScheduling("123");
+
+      assertTrue(booleanCompletableFuture.isDone());
+      assertTrue(booleanCompletableFuture.get());
+      verify(scheduledFutureMock).cancel(false);
+      assertEquals(0, expirationScheduler.countScheduledRequests());
+   }
+
+   @Test
+   public void testAbortSchedulingShouldNotWorkIfRequestIsNotCompleted() throws Exception {
+
+      expirationScheduler.scheduleForCompletion("123", booleanCompletableFuture, 42, TimeUnit.SECONDS);
+      RequestExpirationScheduler.ScheduledRequest scheduledRequest = expirationScheduler.get("123");
+      assertEquals(1, expirationScheduler.countScheduledRequests());
+
+      expirationScheduler.abortScheduling("123");
```

Comment:
This test to me seems like a bug if I look at it just from the viewpoint of a service, since if I call `abortScheduling` I would expect it to abort the scheduled service, like what I said in the previous comment. Just saying :)

Reply:
ok, I will change that

Reply:
I see that you added an additional argument, which is never used :P Not what I was thinking, but like I said I am fine with it.

Reply:
I agreed with you on the point of not being well documented and API side. I'm know is not used now, but I'm leaving soon for some months and I was wondering if somebody else might need to check and change this after. So this is why I added both ways for now. thanks for the merge!!!